### PR TITLE
Update eslint to latest and add linting to build

### PR DESCRIPTION
**Pull request by knolleary**
Updates eslint to our standard version and includes linting in the build

### CODE REVIEW

**Comment by knolleary** [comment]
Note I have not updated `eslint-plugin-vue` which does have a major version bump available. The linting passes without it - but we should look at what were missing with that update.

File: package.json [hunk 1]
```diff
@@ -53,7 +53,8 @@
     "scripts": {
         "serve": "vue-cli-service serve",
         "build": "rollup -c",
-        "build-docs": "vue-cli-service build"
+        "build-docs": "vue-cli-service build",
+        "lint": "eslint -c .eslintrc \"src/**/*.js\""
```
Comment (Pezmc) [inline package.json:57]: I believe this should just be `src/` and `.vue` files

i.e: `eslint --ext .js,.vue src/`

Comment (Pezmc) [inline package.json:57]:
```suggestion
        "lint": "eslint --ext .js,.vue src/"
```

[the remark this answers is not in the document]
Comment (Pezmc) [reply package.json]: On third (and hopefully final thoughts), we should just drop the `vue-cli-service` dependency entirely (it's [deprecated](https://cli.vuejs.org)) and use my suggestion above.

**Review by Pezmc — APPROVED** [review]

**Pull request merged**